decks/promises/01-promise-chain.mdx

```diff
@@ -0,0 +1,198 @@
+import { Head, Appear, Image, withDeck } from 'mdx-deck'  // https://github.com/jxnblk/mdx-deck
+import { CodeSurfer } from "mdx-deck-code-surfer"
+import { FullScreenCode } from "mdx-deck/layouts";
+import nightOwl from "prism-react-renderer/themes/nightOwl"
+
+## Promise Chaining
+
+- A `Promise` returns a `Promise`
+- Once inside of a `.then` - the return value is turned into a promise, if it isn't already
+- You can chain as many `.then` as you need 
+
+---
+<CodeSurfer
+  code={require("raw-loader!./snippets/promise-chain.snippet")}
+  theme={nightOwl}
+  lang="javascript"
+  title="Promise Chains"
+  showNumbers
+  steps={[
+    {}, // First step should be an overview of the snippet
+    {lines: [2], notes: "This returns a promise"},
+    {range: [6,8], notes: "Once inside a then - it wraps the result in a promise"},
+    {range: [9,11], notes: "Allowing us to chain operations together"},
+    {range: [13,21], notes: "Similar to the managers"},
+    {range: [17,18], notes: "But easy to add new features / requirements"},
+    {}
+  ]}
+/>
+---
+
+### Keeping these functions simple can help identify reusable patterns
+
+<ul style={{listStyle: 'none'}}>
+<Appear>
+    <li>Filtering by a user type</li>
+    <li>Calculating the total of a value</li>
+</Appear>
+</ul>
+
+---
+<CodeSurfer
+  code={require("raw-loader!./snippets/promise-chain-2.snippet")}
+  theme={nightOwl}
+  lang="javascript"
+  title="Extracting Reusability"
+  showNumbers
+  steps={[
+    {range: [5,8], notes: 'Extract Reusable Logic'}, 
+    {range: [11,12], notes: 'Use in Promise Chain'}, 
+    {range: [15,17], notes: 'Reuse same functions in new context'}, 
+    {}
+  ]}
+/>
+
+---
+# Promise Patterns  
+---
+
+## Promise Patterns - Parallel
+
+- Finish in any order
+- Promises do not depend on the success of another 
+- Only care once they all finish 
+
+---
+
+## Promise Patterns - Waterfall
+
+- Order matters
+- Another request needs the result of a previous 
+
+---
+
+## Promise Parallel - Promise.all
+
+* Waits for all promises to resolve
+* Gets passed into the handler as an array
+
+```javascript
+let promiseA = someRequest('A');
+let promiseB = someRequest('B');
+let PromiseC = someRequest('C');
+
+Promise.all([promiseA,promiseB,promiseC])
+    .then(results=>{
+        console.log('Hello Results',results)
+    })
+```
```

Comment:
This block should be separated as it runs off the screen a little.
I think you should also note here that this usually needs to be polyfilled (we recommend Bluebird)

Reply:
cleaned up the spacing - will mention the polyfill (so used to tooling adding this for us that I don't think about it)